kroxylicious-systemtests/src/test/java/io/kroxylicious/systemtests/OperatorChangeDetectionST.java

```diff
@@ -89,12 +119,31 @@ void setUp(String namespace) {
 
     @AfterAll
     void cleanUp() {
-        kroxyliciousOperator.delete();
+        if (kroxyliciousOperator != null) {
+            kroxyliciousOperator.delete();
+        }
     }
 
     @BeforeAll
     void setupBefore() {
         kroxyliciousOperator = new KroxyliciousOperator(Constants.KROXYLICIOUS_OPERATOR_NAMESPACE);
         kroxyliciousOperator.deploy();
     }
+
+    private String getInitialChecksum(String namespace, KubeClient kubeClient) {
+        await().atMost(Duration.ofSeconds(90)).untilAsserted(() -> {
```

Comment:
You should should be able to use org.awaitility.core.ConditionFactory#untilAsserted(java.util.concurrent.Callable<T>, java.util.function.Consumer<? super T>) to avoid the need to re-list the pods once the condition comes true.